### DISCUSSION TITLE
Close to Close Volatility

### DISCUSSION THREAD

**Discussion by wclark17** [DaveSkender/Stock.Indicators#1010] [Ideas]
Any plans to add this volatility metric?
Although I came across a few different methods for the formula, but it does seem like it could be useful

**Comment by DaveSkender** [comment]
Never heard of it, but maybe it’s just a name thing.  What’s the gist?

**Comment by wclark17** [comment]
It tries to measure how volatile recent trading has been by comparing the
closes over X periods.  This is the one I coded up:

https://portfolioslab.com/tools/close-to-close-volatility

There are a few different formulas but the others are looking to annualized
the return and look at daily closes versus using the closes of different ca
dle sizes.  Like this one
https://www.macroption.com/historical-volatility-excel/

**Reply by DaveSkender** [reply]
Oh, this is just another name for our [standard deviation](https://dotnet.stockindicators.dev/indicators/StdDev/#content) indicator: `quotes.GetStdDev(..);`.  Here’s the [underlying code](https://github.com/DaveSkender/Stock.Indicators/blob/7020552f8427cdede48270907c84ab800d5f1579/src/_common/Math/Numerix.cs#L5) for our calculations, for comparison.  Definitely a classic.

**Reply by DaveSkender** [reply]
The only variant here, as you pointed out, might be to use return values (% change in price) instead of raw close price, which is a good way to normalize across assets.  You can do that fairly simply with a chain from daily [rate of change](https://dotnet.stockindicators.dev/indicators/Roc/#content):

```csharp
var results = quotes
  .GetRoc(1)
  .GetStdDev(..):
```

**Reply by wclark17** [reply]
Thanks. for taking a look.  
I'll play with that chain, as it makes sense.

One question is how would you annualize it?  Although I haven't been able to make the connection with annualizing when using say a lookback of 10 on 15m candles.

**Reply by DaveSkender** [reply]
Aw, man, I have to crack open my old schoolbooks for annualizing returns.  That second link you have above uses a natural log based approach, but that looks goofy to me.
